Source/DataSources/ModelVisualizer.js

```diff
@@ -364,8 +375,68 @@ ModelVisualizer.prototype.getBoundingSphere = function (entity, result) {
   }
 
   const hasHeightReference = model.heightReference !== HeightReference.NONE;
-  if (hasHeightReference && !defined(model._clampedModelMatrix)) {
-    return BoundingSphereState.PENDING;
+  if (hasHeightReference) {
+    const scene = this._scene;
+    const globe = scene.globe;
+    const ellipsoid = globe.ellipsoid;
+    const terrainProvider = globe.terrainProvider;
+
+    const modelMatrix = model.modelMatrix;
+    scratchPosition.x = modelMatrix[12];
+    scratchPosition.y = modelMatrix[13];
+    scratchPosition.z = modelMatrix[14];
+    const cartoPosition = Ellipsoid.WGS84.cartesianToCartographic(
+      scratchPosition
+    );
+
+    // Since the model's bounding sphere may be clamped to a lower LOD tile if
+    // the camera is initially far away, we use sampleTerrainMostDetailed to estimate
+    // where the bounding sphere should be and set that as the target bounding sphere
+    // for the camera.
+    let clampedBoundingSphere = this._modelHash[entity.id]
+      .clampedBoundingSphere;
+
+    if (!defined(clampedBoundingSphere)) {
+      clampedBoundingSphere = new BoundingSphere();
+
+      // Since this function is called per-frame, we set a flag when sampleTerrainMostDetailed
+      // is called and check for it to avoid calling it again.
+      const awaitingSampleTerrain = this._modelHash[entity.id]
+        .awaitingSampleTerrain;
+      if (!awaitingSampleTerrain) {
+        Cartographic.clone(cartoPosition, scratchCartographic);
+        this._modelHash[entity.id].awaitingSampleTerrain = true;
+        ModelVisualizer._sampleTerrainMostDetailed(terrainProvider, [
+          scratchCartographic,
+        ]).then((result) => {
+          this._modelHash[entity.id].awaitingSampleTerrain = false;
+
+          const updatedCartographic = result[0];
+          if (model.heightReference === HeightReference.RELATIVE_TO_GROUND) {
+            updatedCartographic.height += cartoPosition.height;
+          }
+          ellipsoid.cartographicToCartesian(
+            updatedCartographic,
+            scratchPosition
+          );
+
+          // Update the bounding sphere with the updated position.
+          BoundingSphere.clone(model.boundingSphere, clampedBoundingSphere);
+          clampedBoundingSphere.center = scratchPosition;
+
+          this._modelHash[
+            entity.id
+          ].clampedBoundingSphere = BoundingSphere.clone(clampedBoundingSphere);
+        });
+      }
```

Comment:
Shouldn't there be a catch with error handling here? Otherwise it is just stuck in awaiting forever, isn't it?